Automattic/sensei
# Add Slack notification when tests fail in trunk

## Proposed Changes
This PR adds support for sending Slack notifications to the `sensei-janitorial` Slack channel when any PHP, JS or E2E test fails in `trunk`.

In my testing, I noticed that sometimes multiple notifications are sent in the case of a PHP test failure. Although setting `fail-fast` to `true` seems to mitigate this, there could still be occasional instances of multiple PHP failure notifications. This appears to be related to how close to completion a given job is, as they all run in parallel.

See p1743537803983899/1743533786.323999-slack-C02NWDZBL0H to maintain the Slack app.

UPDATE: As of April 23, 2024, the value of `SLACK_JANITORIAL_CHANNEL` was changed from `sensei-janitorial` to `sensei` so that failures will be sent to the `#sensei` channel going forward. 

## Testing Instructions


The true test will be after we merge this PR to `trunk` and see test failures being sent to the channel, but you can also try the following to ensure notifications aren't sent from PRs:

1. Add a commit to this branch that causes JS, E2E and PHP test failures.
2. Verify that no notifications are sent to the `sensei-janitorial` channel (since they are only sent on a `push` event, which only happens in [`trunk.yml`](https://github.com/Automattic/sensei/blob/trunk/.github/workflows/trunk.yml#L3)).
3. You can also verify that JS Testing, PHP Testing and E2E Testing notifications were previously sent to the `sensei-janitorial` channel by this PR when I was https://github.com/Automattic/sensei/pull/7030/commits/076b58dc04a45d8bdc7414d5a0a5dd9fe92f74c9#diff-3e103440521ada06efd263ae09b259e5507e4b8f7408308dc227621ad9efa31eR3.

![Screenshot 2023-07-18 at 12 18 48 PM](https://github.com/Automattic/sensei/assets/1190420/24a491a7-ae92-484a-90ce-18fba6119929)

## Pre-Merge Checklist
